packages/amplify-category-api/src/provider-utils/awscloudformation/PipelineWithAwaiter.ts

```diff
@@ -0,0 +1,380 @@
+      timeout: cdk.Duration.seconds(25),
+      code: lambda.Code.fromInline(isCompleteHandlerCode),
+    });
+    isCompleteHandler.addToRolePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ['codepipeline:GetPipeline', 'codepipeline:ListPipelineExecutions'],
+        resources: [pipelineArn],
+      }),
+    );
+    isCompleteHandler.addToRolePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ['cloudformation:DescribeStacks'],
+        resources: [cdk.Stack.of(scope).stackId],
+      }),
+    );
+
+    const myProvider = new custom.Provider(scope, `${id}MyProvider`, {
+      onEventHandler,
+      isCompleteHandler,
+      queryInterval: cdk.Duration.seconds(10),
+    });
+
+    const customResource = new cdk.CustomResource(scope, `Deployment${id}`, {
+      serviceToken: myProvider.serviceToken,
+      properties: {
+        artifactBucketName,
+        artifactKey,
+        pipelineName,
+        deploymentMechanism,
+      },
+    });
+
+    super(scope, id);
+  }
+}
+
+export class PipelineWithAwaiter extends cdk.Construct {
+  constructor(
+    scope: cdk.Construct,
+    id: string,
+    {
+      bucket,
+      s3SourceActionKey,
+      service,
+      deploymentMechanism,
+      githubSourceActionInfo,
+      containersInfo,
+      desiredCount,
+    }: {
+      bucket: s3.IBucket;
+      s3SourceActionKey?: string;
+      deploymentMechanism: DEPLOYMENT_MECHANISM;
+      githubSourceActionInfo?: GitHubSourceActionInfo;
+      service: ecs.CfnService;
+      containersInfo: {
+        container: ecs.ContainerDefinition;
+        repository: ecr.Repository;
+      }[];
+      desiredCount: number;
+    },
+  ) {
+    super(scope, id);
+
+    const sourceOutput = new codepipeline.Artifact('SourceArtifact');
+    const buildOutput = new codepipeline.Artifact('BuildArtifact');
+
+    const codebuildproject = new codebuild.PipelineProject(scope, `${id}CodeBuildProject`, {
+      environment: {
+        buildImage: codebuild.LinuxBuildImage.STANDARD_4_0,
+        // See: https://docs.aws.amazon.com/codebuild/latest/userguide/troubleshooting.html#troubleshooting-cannot-connect-to-docker-daemon
+        privileged: true,
+      },
+    });
+
+    if (githubSourceActionInfo && githubSourceActionInfo.tokenSecretArn) {
+      codebuildproject.addToRolePolicy(
+        new iam.PolicyStatement({
+          effect: iam.Effect.ALLOW,
+          actions: [
+            'secretsmanager:GetRandomPassword',
+            'secretsmanager:GetResourcePolicy',
+            'secretsmanager:GetSecretValue',
+            'secretsmanager:DescribeSecret',
+            'secretsmanager:ListSecretVersionIds',
+          ],
+          resources: [githubSourceActionInfo.tokenSecretArn],
+        }),
+      );
+    }
+
+    if (codebuildproject.role) {
+      codebuildproject.role.addToPrincipalPolicy(
+        new iam.PolicyStatement({
+          resources: ['*'],
```

Comment:
Is it possible to scope this down from a "*"?

Reply:
@kaustavghosh06 no its not, we confirmed with the ECR team this is necessary